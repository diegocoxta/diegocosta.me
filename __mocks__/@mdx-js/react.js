module.exports = {
  MDXProvider: jest.fn(({ children }) => children),
};
