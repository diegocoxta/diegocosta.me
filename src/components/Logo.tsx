import React from 'react';
import styled from 'styled-components';

import { Link } from '@app/components/LanguageSwitcher';

export interface BrandNameProps {
  author: string;
  href: string;
  size: 'small' | 'large';
}

const Container = styled(Link)`
  text-decoration: none;
  box-shadow: none;
  display: flex;
  align-items: center;

  &:focus {
    outline: none;
  }
`;

const Name = styled.h1<{ size: BrandNameProps['size'] }>`
  font-size: ${(props) => (props.size === 'large' ? '15vw' : '6vw')};
  color: ${({ theme }) => theme.titleColor};
  font-weight: 700;
  text-transform: lowercase;
  margin: 0;

  @media (min-width: 760px) {
    font-size: ${(props) => (props.size === 'large' ? '110px' : '40px')};
  }
`;

const LastName = styled.span`
  color: ${({ theme }) => theme.accentColor};
`;

export default function BrandName(props: BrandNameProps) {
  const [name, lastname] = props.author.split(' ');

  return (
    <Container to={props.href}>
      <Name size={props.size}>
        {name}
        {lastname && <LastName data-testid="header-lastname">{lastname[0]}.</LastName>}
      </Name>
    </Container>
  );
}

BrandName.defaultProps = {
  size: 'large',
  href: '/',
};
