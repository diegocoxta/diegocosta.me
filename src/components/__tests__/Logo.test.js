import React from 'react';
import { render } from '@testing-library/react';

import Logo from '../Logo';

describe('<Logo />', () => {
  it('renders properly', () => {
    const { baseElement, queryByText, queryByTestId } = render(
      <Logo name="Diego Costa" to="https://github.com/diegocoxta" />
    );
    expect(queryByText('Diego')).toBeTruthy();
    expect(queryByText('C.')).toBeTruthy();
    expect(queryByTestId('logo-link').href).toBe('https://github.com/diegocoxta');
    expect(baseElement).toMatchSnapshot();
  });

  it('do not render the last name if not defined', () => {
    const { baseElement, queryByText, queryByTestId } = render(<Logo name="Diego" />);
    expect(queryByText('Diego')).toBeTruthy();
    expect(queryByText('C.')).toBeFalsy();
    expect(queryByTestId('logo-link').href).toBe('http://localhost/');
    expect(baseElement).toMatchSnapshot();
  });
});
