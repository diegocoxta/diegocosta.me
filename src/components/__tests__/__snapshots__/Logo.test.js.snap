// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> do not render the last name if not defined 1`] = `
.c0 {
  text-decoration: none;
  box-shadow: none;
  display: flex;
  align-items: center;
}

.c0:focus {
  outline: none;
}

.c1 {
  font-size: 15vw;
  font-weight: 700;
  text-transform: lowercase;
  margin: 0;
}

@media (min-width: 760px) {
  .c1 {
    font-size: 110px;
  }
}

<body>
  <div>
    <a
      class="c0"
      data-testid="logo-link"
      href="/"
    >
      <h1
        class="c1"
      >
        Diego
      </h1>
    </a>
  </div>
</body>
`;

exports[`<Logo /> renders properly 1`] = `
.c0 {
  text-decoration: none;
  box-shadow: none;
  display: flex;
  align-items: center;
}

.c0:focus {
  outline: none;
}

.c1 {
  font-size: 15vw;
  font-weight: 700;
  text-transform: lowercase;
  margin: 0;
}

@media (min-width: 760px) {
  .c1 {
    font-size: 110px;
  }
}

<body>
  <div>
    <a
      class="c0"
      data-testid="logo-link"
      href="https://github.com/diegocoxta"
    >
      <h1
        class="c1"
      >
        Diego
        <span
          class=""
          data-testid="logo-lastname"
        >
          C
          .
        </span>
      </h1>
    </a>
  </div>
</body>
`;
